cli/verifier/image_ref_valid.go

```diff
@@ -0,0 +1,62 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package verifier
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/edgelesssys/contrast/internal/kuberesource"
+	"github.com/google/go-containerregistry/pkg/name"
+)
+
+// ImageRefValid verifies that all image references contain valid tag and digest.
+type ImageRefValid struct{}
+
+// Verify verifies that neither the tag nor the digest of image references are empty.
+func (v *ImageRefValid) Verify(toVerify any) error {
+	var findings error
+
+	resources, err := kuberesource.ResourcesToUnstructured([]any{toVerify})
+	if err != nil {
+		return err
+	}
+
+	var images []string
+	for _, r := range resources {
+		findImageFields(r.Object, &images)
+	}
+
+	for _, img := range images {
+		if _, err := name.NewDigest(img); err != nil {
+			findings = errors.Join(findings, fmt.Errorf("the image reference '%s' could not be verified: %w", img, err))
```

Comment:
now it is unclear what the issue is in case only the digest is missing, isn't it? These checks should also output something that is actionable by a user.